server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -104,6 +108,30 @@ public void uploadBlobs(
 
     }
 
+    // Builds a metadata map containing the Base64-encoded checkpoint file data associated with a translog file.
+    static Map<String, String> buildTransferFileMetadata(InputStream metadataInputStream) throws IOException {
+        Map<String, String> metadata = new HashMap<>();
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+            byte[] buffer = new byte[128];
+            int bytesRead;
+            int totalBytesRead = 0;
+
+            while ((bytesRead = metadataInputStream.read(buffer)) != -1) {
+                byteArrayOutputStream.write(buffer, 0, bytesRead);
+                totalBytesRead += bytesRead;
+                if (totalBytesRead > 1024) {
```

Comment:
Why do we need this check?

Reply:
I understand that but why we added it?

Reply:
We added it as a safety measure to prevent excessive memory usage, especially if the checkpoint file is unexpectedly large or corrupted..

Reply:
If the checkpoint file is unexpectedly large, we still need to read it, right? My concern is: if in newer versions, checkpoint file gets bigger, we will start failing the reads. Not sure how we decided on the `1024` bytes limit.

Also, let's define it as a constant instead of hardcoding in the code.

Reply:
During creation of checkpoint object we have a  a size limit: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/translog/Checkpoint.java#L263